spec/lib/middleware/websocket/websocket_connection_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+RSpec.describe WebsocketConnection do
+  let(:app) { double }
+  let(:websocket_connection) { WebsocketConnection.new(app) }
+  let(:env) { { 'rack.upgrade?' => rack_upgrade?, 'rack.upgrade' => nil } }
+
+  describe '#call' do
+    subject { websocket_connection.call(env) }
+
+    context 'when rack.upgrade? is nil' do
+      let(:rack_upgrade?) { nil }
+
+      it 'routes past the websocket middleware' do
+        expect(app).to receive(:call).with(env)
+        subject
+      end
+    end
+
+    context 'when rack.upgrade? is a websocket' do
```

Comment:
⚒ Slight nitpick on the language. The environment variable isn't a web socket, it contains a label (`:websocket`).

Reply:
fixed.